test/Common.cpp

```diff
@@ -45,6 +51,66 @@ boost::filesystem::path discoverTestPath()
 	return {};
 }
 
+std::string IPCEnvOrDefaultPath()
+{
+	if (auto path = getenv("ETH_TEST_IPC"))
+		return path;
+
+	if (auto home = getenv("HOME"))
+		return std::string(home) + "/.ethereum/geth.ipc";
+
+	return std::string{};
+}
+
+CommonOptions::CommonOptions(std::string _caption):
+	options(_caption,
+		po::options_description::m_default_line_length,
+		po::options_description::m_default_line_length - 23
+	)
+{
+	options.add_options()
+		("testpath", po::value<fs::path>(&this->testPath)->default_value(dev::test::testPath()), "path to test files")
+		("ipcpath", po::value<fs::path>(&ipcPath)->default_value(IPCEnvOrDefaultPath()), "path to ipc socket")
+		("no-ipc", po::bool_switch(&disableIPC), "disable semantic tests")
+		("no-smt", po::bool_switch(&disableSMT), "disable SMT checker");
+}
+
+void CommonOptions::validate() const
+{
+	solAssert(
```

Comment:
All four assertions in this function are not really assertions. They are checks that should throw an exception, but it is not an "InternalCompilerError" if they fail. Please either use ifs and throw or solAssertThrow and a different exception.